src/main/java/org/broadinstitute/hellbender/tools/picard/sam/CreateSequenceDictionary.java

```diff
@@ -63,6 +66,11 @@
         if (URI == null) {
             URI = "file:" + REFERENCE_SEQUENCE.getAbsolutePath();
         }
+        if (OUTPUT == null) {
+            // determine the name for the dict file in the same way as CachingIndexedFastaSequenceFile.checkAndCreate
+            final String fastaExt = REFERENCE_SEQUENCE.getAbsolutePath().endsWith("fa") ? "\\.fa$" : "\\.fasta$";
```

Comment:
While I was doing the same in Picard, I realized that I could support all the `FASTA_EXTENSIONS` instead of only "fa" and "fasta". The last commit contains this solution.